calico_containers/integrations/kubernetes/tests/kube_plugin_test.py

```diff
@@ -0,0 +1,414 @@
+
+    def test_create(self):
+        with patch.object(self.plugin, '_configure_interface',
+                          autospec=True) as m_interface, \
+                patch.object(self.plugin, '_configure_profile',
+                             autospec=True) as m_prof:
+            # Set up mock objects
+            m_interface.return_value = 'endpt_id'
+
+            # Set up args
+            args = ['script name','setup','','pod-name','dockerId']
+
+            # Call method under test
+            self.plugin.create(args)
+
+            # Assert
+            self.assertEqual('pod_name', self.plugin.pod_name)
+            self.assertEqual('dockerId', self.plugin.docker_id)
+            m_interface.assert_called_once_with()
+            m_prof.assert_called_once_with('endpt_id')
+
+    def test_create_fail(self):
+        with patch.object(self.plugin, '_configure_interface',
+                          autospec=True) as m_interface, \
+                patch.object(self.plugin, '_configure_profile',
+                             autospec=True) as m_prof, \
+                patch('calico_containers.integrations.'
+                      'kubernetes.calico_kubernetes.sys.exit',
+                      autospec=True) as m_sys_exit:
+            # Set up mock objects
+            m_interface.side_effect = CalledProcessError(1,'','')
+
+            # Set up args
+            args = ['script name','setup','','podname','dockerId']
+
+            # Call method under test
+            self.plugin.create(args)
+
+            # Assert
+            m_sys_exit.assert_called_once_with(1)
+
+    def test_delete(self):
+        with patch.object(self.plugin, 'calicoctl',
+                          autospec=True) as m_calicoctl:
+            # Set up args
+            args = ['script name','teardown', '','pod-name','dockerId']
+
+            # Call method under test
+            self.plugin.delete(args)
+
+            # Assert
+            call_1 = call('container', 'remove', 'dockerId')
+            call_2 = call('profile', 'remove', 'pod_name')
+            calls = (call_1,call_2)
+            m_calicoctl.assert_has_calls(calls)
+
+            self.assertEqual(self.plugin.pod_name, 'pod_name')
+            self.assertEqual(self.plugin.docker_id, 'dockerId')
+
+    def test_configure_interface(self):
+        with patch.object(self.plugin, '_read_docker_ip',
+                          autospec=True) as m_read_docker, \
+                patch.object(self.plugin, '_delete_docker_interface',
+                             autospec=True) as m_delete_docker, \
+                patch.object(calico_kubernetes, 'container_add',
+                             autospec=True) as m_add_container, \
+                patch.object(calico_kubernetes, 'generate_cali_interface_name',
+                             autospec=True) as m_cali, \
+                patch.object(self.plugin, '_get_node_ip',
+                             autospec=True) as m_node_ip, \
```

Comment:
Make sure to keep your mock names close to the names of the methods they're mocking out.

m_node_ip -> m_get_node-ip
m_cali -> m_gen_interface_name